### PR TITLE
Added support for IAM credential in the token field and YC_TOKEN env

### PR DESCRIPTION
**Pull request by Direnol**
### Description
Added support for authentication via IAM credential for CLI. Simplifies the use of different profiles.

### Use Cases

- `YC_TOKEN=$(yc iam create-token) packer build image.json`
- via filed *token*

### CODE REVIEW

File: website/pages/docs/builders/yandex.mdx [hunk 1]
```diff
@@ -21,13 +21,16 @@ based on existing images.
 Builder can authenticate with Yandex.Cloud using one of the following methods:
 
 - OAuth token
+- IAM credential
```
Comment (GennadySpb) [inline website/pages/docs/builders/yandex.mdx:24]:
```suggestion
- IAM token
```

File: website/pages/docs/builders/yandex.mdx [hunk 2]
```diff
@@ -21,13 +21,16 @@ based on existing images.
...
 - File with Service Account Key
 - Service Account assigned to Compute Instance
 
 ### Authentication Using Token
 
 To authenticate with an OAuth token only `token` config key is needed.
-Or use the `YC_TOKEN` environment variable with proper value. Token you could get [here](https://oauth.yandex.ru/authorize?response_type=token&client_id=1a6990aa636648e9b2ef855fa7bec2fb).
+Or use the `YC_TOKEN` environment variable with proper value.
+Token you could get [here](https://oauth.yandex.ru/authorize?response_type=token&client_id=1a6990aa636648e9b2ef855fa7bec2fb).
+Check [documentation](https://cloud.yandex.ru/docs/iam/operations/iam-token/create) about how to get IAM credential.
```
Comment (GennadySpb) [inline website/pages/docs/builders/yandex.mdx:33]:
```suggestion
Check [documentation](https://cloud.yandex.ru/docs/iam/operations/iam-token/create) about how to create IAM token.
```

**Review by GennadySpb — CHANGES_REQUESTED** [review]

File: builder/yandex/driver_yc.go [hunk 3]
```diff
@@ -84,6 +89,7 @@ func NewDriverYC(ui packer.Ui, ac *AccessConfig) (Driver, error) {
 		grpc.WithDefaultCallOptions(grpc.Header(&userAgentMD)),
 		grpc.WithUnaryInterceptor(interceptorChain))
 
+	log.Printf("%v", sdk.IAM())
```
[the remark this answers is not in the document]
Comment (Direnol) [reply builder/yandex/driver_yc.go:92]: Miss

**Comment by GennadySpb** [comment]
LGTM

/cc @azr @SwampDragons

**Pull request merged**